rules/windows/defense_evasion_posh_obfuscation_char_arrays.toml

```diff
@@ -0,0 +1,106 @@
+[metadata]
+creation_date = "2025/04/14"
+integration = ["windows"]
+maturity = "production"
+min_stack_comments = "The ES|QL MATCH operator was introduced in 8.17"
+min_stack_version = "8.17.0"
+updated_date = "2025/04/14"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies PowerShell scripts that use character arrays and runtime string reconstruction as a form of obfuscation. This
+technique breaks strings into individual characters, often using constructs like char[] with index-based access or
+joining logic. These methods are designed to evade static analysis and bypass security protections such as the
+Antimalware Scan Interface (AMSI).
+"""
+from = "now-9m"
+language = "esql"
+license = "Elastic License v2"
+name = "Potential PowerShell Obfuscation via Character Array Reconstruction"
+risk_score = 21
+rule_id = "85e2d45e-a3df-4acf-83d3-21805f564ff4"
+setup = """## Setup
+
+The 'PowerShell Script Block Logging' logging policy must be enabled.
+Steps to implement the logging policy with Advanced Audit Configuration:
+
+```
+Computer Configuration >
+Administrative Templates >
+Windows PowerShell >
+Turn on PowerShell Script Block Logging (Enable)
+```
+
+Steps to implement the logging policy via registry:
+
+```
+reg add "hklm\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging" /v EnableScriptBlockLogging /t REG_DWORD /d 1
+```
+"""
+severity = "low"
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Defense Evasion",
+    "Data Source: PowerShell Logs",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+FROM logs-windows.powershell_operational* metadata _id, _version, _index
+| WHERE event.code == "4104"
+
+// Filter for scripts that contain the "char" keyword using MATCH, boosts the query performance
+| WHERE powershell.file.script_block_text : "char"
```

Comment:
Did you mean `powershell.file.script_block_text == "char"`

Reply:
Looked at the docs, did you mean to use `Match()`?

https://www.elastic.co/docs/reference/query-languages/esql/functions-operators/search-functions#esql-match